File: modelexpress_server/src/registry.rs
// SPDX-FileCopyrightText: Copyright (c) 2025-2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

//! Distributed model registry: model-download lifecycle (`DOWNLOADING` / `DOWNLOADED` /
//! `ERROR`) and LRU cache-eviction timestamps.
//!
//! `backend` owns the `RegistryBackend` trait plus Redis and Kubernetes CRD
//! implementations. `state` wraps the backend in a lazy-connect manager used by
//! `ModelDownloadTracker` and `CacheEvictionService`.

pub mod backend;
pub mod k8s_types;
pub mod state;

File: modelexpress_client/python/modelexpress/quantization_providers/registry.py
# SPDX-FileCopyrightText: Copyright (c) 2025-2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
# SPDX-License-Identifier: Apache-2.0

"""Registry for quantization RDMA manifest providers."""

from __future__ import annotations

from .base import DefaultManifestProvider, QuantizationManifestProvider
from .humming import HummingManifestProvider

_DEFAULT_PROVIDER = DefaultManifestProvider()
_PROVIDERS: tuple[QuantizationManifestProvider, ...] = (
    HummingManifestProvider(),
)


def get_quantization_provider(quantization: str = "") -> QuantizationManifestProvider:
    for provider in _PROVIDERS:
        if provider.enabled(quantization):
            return provider
    return _DEFAULT_PROVIDER
